### PR TITLE
Optional opencv

### LINKED ISSUE #31: Ginga widget: Make opencv truly optional
It works without ``opencv``, just slower when rendering image rotation and such.

- [ ] Make it truly optional in widget constructor.
- [ ] Remove it from CI dependencies and setup. (Removed from setup in #33)
- [ ] Make it optional in installation doc.

### PR DESCRIPTION
**Pull request by pllim**
Fix #31 

Also moved dev notes to https://github.com/astropy/astrowidgets/wiki/DevNotes and added link to example notebook to docs.

### CODE REVIEW

**Comment by pllim** [comment]
Huh... Will have to investigate failure later... https://travis-ci.org/astropy/astrowidgets/jobs/420294263

File: astrowidgets/core.py [hunk 1]
```diff
@@ -97,15 +46,19 @@ class ImageWidget(ipyw.VBox):
...
         super().__init__()
+
         # TODO: Is this the best place for this?
         if use_opencv:
-            from ginga import trcalc
-            trcalc.use('opencv')
+            try:
+                from ginga import trcalc
+                trcalc.use('opencv')
+            except (ImportError, ModuleNotFoundError) as exc:
```
Comment (mwcraig) [inline astrowidgets/core.py]: Looks like `ModuleNotFoundError` is Python 3.6 and up only...but it subclasses from `ImportError` so just catching `ImportError` should be enough.

**Comment by pllim** [comment]
Thanks for the help, @mwcraig ! Tests pass now. 🎉

**Comment by mwcraig** [comment]
Thanks, merging! Hoping to put some concentrated time into this later this week/weekend.

**Pull request merged**